odxtools/basicstructure.py

```diff
@@ -247,26 +246,6 @@ def decode_from_pdu(self, decode_state: DecodeState) -> ParameterValue:
 
         return result
 
-    def encode(self, coded_request: Optional[bytes] = None, **kwargs: ParameterValue) -> bytes:
```

Comment:
I need the encode function on the structure itself:
Use case: I split services into static parts an dynamic parts (that are mostly structures), and use the dynamic part to generate `udsoncan` configuration as per https://udsoncan.readthedocs.io/en/latest/udsoncan/examples.html#reading-a-did-with-readdatabyidentifier

Reply:
can't you use `.encode_into_pdu()`? (creating an `EncodeState` object and calling `self.encode_into_pdu()` all `.encode()` does.) The reason I would like to move the toplevel function to the Request/Response classes is that they require different parameters, i.e., `Response.encode()` needs `triggering_request` which does not make sense for requests...